guides/keras_cv/object_detection_keras_cv.py

```diff
@@ -0,0 +1,846 @@
+        num_parallel_calls=tf.data.AUTOTUNE,
+    )
+    return ds
+
+
+train_ds = load_pascal_voc(
+    split="train", dataset="voc/2007", bounding_box_format="xywh"
+)
+eval_ds = load_pascal_voc(split="test", dataset="voc/2007", bounding_box_format="xywh")
+eval_ds = load_pascal_voc(dataset="voc/2007", split="test", bounding_box_format="xywh")
+
+train_ds = train_ds.shuffle(BATCH_SIZE * 4)
+
+"""
+Next, let's batch our data.
+
+In KerasCV object detection tasks it is recommended that
+users use ragged batches of inputs.
+This is due to the fact that images may be of different sizes in PascalVOC,
+as well as the fact that there may be different numbers of bounding boxes per
+image.
+
+To construct a ragged dataset in a `tf.data` pipeline, you can use the
+`ragged_batch()` method.
+"""
+
+train_ds = train_ds.ragged_batch(BATCH_SIZE, drop_remainder=True)
+eval_ds = eval_ds.ragged_batch(BATCH_SIZE, drop_remainder=True)
+
+"""
+Let's make sure our dataset is following the format KerasCV expects.
+By using the `visualize_dataset()` function, you can visually verify
+that your data is in the format that KerasCV expects.  If the bounding boxes
+are not visible, or are visible in the wrong locations that is a sign that your
+data is mis-formatted.
+"""
+
+visualize_dataset(
+    train_ds, bounding_box_format="xywh", value_range=(0, 255), rows=2, cols=2
+)
+
+"""
+And for the eval set:
+"""
+
+visualize_dataset(
+    eval_ds,
+    bounding_box_format="xywh",
+    value_range=(0, 255),
+    rows=2,
+    cols=2,
+    # If you are not running your experiment on a local machine, you can also
+    # make `visualize_dataset()` dump the plot to a file using `path`:
+    # path="eval.png"
+)
+
+"""
+Looks like everything is structured as expected.
+Now we can move on to constructing our
+data augmentation pipeline.
+
+## Data augmentation
+
+One of the most challenging tasks when constructing object detection
+pipelines is data augmentation.  Image augmentation techniques must be aware of the underlying
+bounding boxes, and must update them accordingly.
+
+Luckily, KerasCV natively supports bounding box augmentation with its extensive
+library
+of [data augmentation layers](https://keras.io/api/keras_cv/layers/preprocessing/).
+The code below loads the Pascal VOC dataset, and performs on-the-fly bounding box
+friendly data augmentation inside of a `tf.data` pipeline.
+"""
+augmenter = keras.Sequential(
+    layers=[
+        keras_cv.layers.RandomFlip(mode="horizontal", bounding_box_format="xywh"),
+        keras_cv.layers.JitteredResize(
+            target_size=(640, 640), scale_factor=(0.75, 1.3), bounding_box_format="xywh"
+        ),
+    ]
+)
+
+train_ds = train_ds.map(augmenter, num_parallel_calls=tf.data.AUTOTUNE)
+visualize_dataset(
+    train_ds, bounding_box_format="xywh", value_range=(0, 255), rows=2, cols=2
+)
+
+"""
+Great!  We now have a bounding box friendly data augmentation pipeline.
```

Comment:
When multiple words create a single adjective you should add dashes for clarity:
* state-of-the-art
* out-of-the-box